contracts/test/integration/MultiSubnet.t.sol

```diff
@@ -127,6 +127,105 @@ contract MultiSubnetTest is Test, IntegrationTestBase {
         assertEq(recipient.balance, amount);
     }
 
+    // A bottom up receipt sending from parent to child. The original message is a
+    // bottom up release message, but the execution worked in the parent, creating
+    // a topdown result message from parent to child
+    function testMultiSubnet_Native_OkResultFromParentToChild() public {
+        address caller = address(new MockIpcContract());
+        address recipient = address(new MockIpcContractPayable());
+        uint256 amount = 3;
+
+        vm.deal(address(rootNativeSubnetActor), DEFAULT_COLLATERAL_AMOUNT);
+        vm.deal(caller, amount);
+        vm.deal(recipient, amount);
+
+        vm.prank(address(rootNativeSubnetActor));
+        registerSubnetGW(DEFAULT_COLLATERAL_AMOUNT, address(rootNativeSubnetActor), rootGateway);
+
+        IpcEnvelope memory crossMsg = CrossMsgHelper.createReleaseMsg(
+            nativeSubnetName,
+            recipient,
+            FvmAddressHelper.from(caller),
+            amount
+        );
+
+        IpcEnvelope memory resultMsg = CrossMsgHelper.createResultMsg(crossMsg, OutcomeType.Ok, new bytes(0));
+        require(resultMsg.value == 0, "ok receipt should have 0 value");
+
+        IpcEnvelope[] memory msgs = new IpcEnvelope[](1);
+        msgs[0] = resultMsg;
+
+        executeTopDownMsgs(msgs, nativeSubnetName, address(nativeSubnetGateway));
+
+        // works with no state changes
+        assertEq(recipient.balance, amount);
```

Comment:
I think this test would be easier to understand this test if it started with 
- recipient/parent having balance=3
- caller/child having balance=0 

This way the reader could imagine that the initial state of the tests is when the child released its balance completely.